package.json

```diff
@@ -10,5 +10,12 @@
     "build:min": "cd code && yarn gulp vscode-reh-web-linux-x64-min",
     "rebuild-native-modules": "cd code && npm rebuild"
   },
-  "license": "EPL-2.0"
-}
+  "license": "EPL-2.0",
+  "dependencies": {
+    "node-gyp": "^9.4.1"
+  },
```

Comment:
in another PR we can see that adding the dependency to the package.json doesn't fix the problem: https://github.com/che-incubator/che-code/pull/305#pullrequestreview-1741612736

Reply:
yes but is it BAD to actually declare your dependencies?

Reply:
it's not a Che-Code-specific dependency - correct?

I see that:
- VS Code doesn't declare it in the package.json file
- adding it as a dependency to the Che-Code doesn't fix the problem

So, I have no idea why we should keep it in the Che-Code package.json file...

Reply:
it's a vscode dependency but they don't have it declared. 

so do you want to push this suggestion upstream? Or just assume that someone will know to globally install node-gyp before they attempt to build code-oss or che-code?

Reply:
> it's a vscode dependency but they don't have it declared.
 so do you want to push this suggestion upstream? 

do you think
> they don't have it declared

by mistake?

Anyway I still don't understand why you trying to add it taking into account that  
> adding it as a dependency to the Che-Code doesn't fix the problem

Reply:
I think they don't have it because until recently node-gyp was bundled and installed in npm 9.5... so maybe they haven't hit the problem we have in https://github.com/che-incubator/che-code/pull/305

Reply:
But you tested it in another PR and it doesn't help to resolve the problem: https://github.com/che-incubator/che-code/pull/305#pullrequestreview-1741612736

You had to apply another solution from Artem: https://github.com/che-incubator/che-code/pull/305#discussion_r1400917560

correct?

and 

Despite this you have merged your change in another PR while we are discussing it here...
before it you broke 7.77.x by merging a PR without approve: https://github.com/che-incubator/che-code/pull/306

something strange...

Reply:
well, the base images update for 7.77.x was merged automatically by script

as to the fix in #305 , I cherrypicked that to 7.77.x so hopefully we're working again in upstream.

Reply:
> If you want to revert

so, 
- someone merge something while it's still discussed 
- then someone revert it? 

I don't think it's an effective way of a collaboration...

Reply:
>  the base images update for 7.77.x was merged automatically by script

I thought that:
-  the script only creates a PR, doesn't it?
- it was created on Nov 18
- and then you manually merged it on Nov 20 with failed jobs

But could confirm that I'm really mistaken and the script automatically merges PRs without taking into account checks for PRs? 
If so, then I guess we will have the same problems in near future and we should improve the script somehow to avoid it...

Reply:
it's generated by tool: https://github.com/eclipse-che/che-release/actions/workflows/update-base-images.yml
and auto-merged by other tool: https://github.com/eclipse-che/che-release/blob/main/utils/approvePRs.sh

I generally don't review them, I just merge them. It saves everyone time :) 

And sometimes it breaks things ... when an update from rhel 8.y to 8.y+1 happens.